src/main.rs

```diff
@@ -74,10 +75,35 @@ fn main() -> Result<i32> {
         true => Policy::limited(args.max_redirects.unwrap_or(10)),
         false => Policy::none(),
     };
+    let timeout = match args.timeout {
+        Some(secs) => {
+            const NANOS_PER_SEC: u32 = 1_000_000_000;
+            let nanos = secs * (NANOS_PER_SEC as f64);
+
+            if !(nanos.is_finite() && nanos.is_sign_positive()) {
+                return Err(anyhow!("Invalid seconds as connection timeout"));
+            }
+            const MAX_TIMEOUT_NANOS: f64 =
+                ((u64::MAX as u128 + 1) * (NANOS_PER_SEC as u128)) as f64;
+            if nanos >= MAX_TIMEOUT_NANOS {
+                return Err(anyhow!(
+                    "Out of range of specifiable seconds as connection timeout"
+                ));
+            }
```

Comment:
We usually do any input validation in `cli.rs`. For this particular case, it would be by defining a newtype that wraps `std::time::Duration` and implementing the `FromStr` trait for it.

```rust
struct Timeout(std::time::Duration);

impl FromStr for Timeout { /* .. */ }

struct Cli {
    #[structopt(long, value_name = "SEC")]
    pub timeout: Option<Timeout>,
}
```

However, It might not be worth the effort of handling all [possible panics](https://doc.rust-lang.org/src/core/time.rs.html#681) from `from_secs_f64` and we could get away by just making sure the value of timeout is not negative.

Reply:
Change to only check that the value has a positive sign. Checking finite value and overflow are disabled.